### PR TITLE
LKE-10209 fix(NodeIcon): Change IMAGE_PATTERN regex to accept  non-whitespace characters at the end

### PR DESCRIPTION
**Pull request by aminerhanemi**

### CODE REVIEW

File: src/styles/nodeAttributes.ts [hunk 1]
```diff
@@ -141,11 +139,11 @@ export class NodeAttributes extends ItemAttributes<INodeStyle> {
             }
           };
         } else if ('image' in style && typeof style.image === 'object') {
+          const urlType = Tools.getType(style.image.url as string);
           result = {
             image: {
               url:
-                Tools.getType(style.image.url as string) === 'imageUrl' ||
-                Tools.getType(style.image.url as string) === 'image'
+                urlType === 'imageUrl' || urlType === 'image'
```
Comment (aminerhanemi) [inline src/styles/nodeAttributes.ts:146]: that doesn't fix the issue of passing a no image URL as Tools.getIn will still throw an error

File: src/tools/ogmaTool.ts [hunk 2]
```diff
@@ -48,8 +48,10 @@ export class OgmaTools {
       return true;
     }
     const hexRegExp = /#[A-Fa-f0-9]{3,6}/;
-    const rgbRegExp = /^rgb\(\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*\)$/i;
-    const rgbaRegExp = /^rgba\(\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*(?:0|1|0?\.\d+)\s*\)$/i;
+    const rgbRegExp =
+      /^rgb\(\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*,\s*([01]?\d\d?|2[0-4]\d|25[0-5])\s*\)$/i;
+    const rgbaRegExp =
```
Comment (hsenbassam) [inline src/tools/ogmaTool.ts:53]: @aminerhanemi I think the main change in the PR is the regex right ?

Comment (aminerhanemi) [reply src/tools/ogmaTool.ts:53]: no, this line is just some formatting. In the end I didn't change the regex

File: src/tools/tools.ts [hunk 3]
```diff
@@ -323,11 +322,16 @@ export class Tools {
    * @return {boolean}
    */
   private static isUrl(value: string): boolean {
```
Comment (aminerhanemi) [inline src/tools/tools.ts:324]: For review

File: src/tools/tools.ts [hunk 4]
```diff
@@ -336,6 +340,20 @@ export class Tools {
     return IMAGE_PATTERN.test(value);
   }
 
+  /**
+   * Return true if the given string is an url with an image extension
+   *
+   * @param {string}  value
+   * @return {boolean}
+   */
+  private static isURLImage(value: string): boolean {
```
Comment (aminerhanemi) [inline src/tools/tools.ts:349]: For review @hsenbassam

**Review by hsenbassam — APPROVED** [review]

**Pull request merged**